universum/modules/code_report_collector.py

```diff
@@ -68,14 +68,14 @@ def _process_one_sarif_issue(self, issue, root_uri_base_paths, who) -> None:
             uri = artifact_data.get('uri')
             if not uri:
                 raise ValueError("Unexpected lack of uri tag")
-            path = urllib.parse.unquote(urllib.parse.urlparse(uri).path)
             if artifact_data.get('uriBaseId'):
                 # means path is relative, need to make absolute
                 uri_base_id = artifact_data.get('uriBaseId', '')
                 root_base_path = root_uri_base_paths.get(uri_base_id, '')
                 if uri_base_id and not root_base_path:
                     raise ValueError(f"Unexpected lack of 'originalUriBaseIds' value for {uri_base_id}")
-                path = str(Path(root_base_path) / path)
+                uri = str(Path(root_base_path) / uri)
```

Comment:
Are you sure you can use `pathlib.Path` to concatenate URIs?
Isn't there a specific function for this in the urllib?